packages/headless/src/features/placement-set/placement-set-interface.ts

```diff
@@ -0,0 +1,40 @@
+export interface Placement {
+  callbackData: string;
+  campaignId: string;
+  visitorId: string;
+}
+
+export interface Badges extends Placement {
+  badges: Badge[];
+}
+
+export interface Badge {
+  id: string;
+  message: string;
+}
+
+export interface Recommendations extends Placement {
+  products: Product[];
+}
+
+export interface Product {
```

Comment:
Can we reuse the standard fields as defined here?
https://github.com/coveo/ui-kit/blob/4579bcef4a080d14498a1ee8c45dd10b8a307493/packages/headless/src/features/fields/fields-state.ts#L48

Reply:
I'll see if can try something to tie those two types together, yeah.

Reply:
I'll be able to do something like: 

```typescript
type Product = {
  [K in typeof StandardCommerceFields[number]]: K extends
    | 'ec_images'
    | 'ec_thumbnails'
    ? string[]
    : string;
} & {
  additionalFields: string;
  childResults: Product[];
  clickUri: string;
  documentUri: string;
  permanentId: string;
  totalNumberOfChildResults: number;
};
```